kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/FeedbackInputFragment.java

```diff
@@ -0,0 +1,215 @@
+        super();
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if (!FeedbackFragmentListener.class.isAssignableFrom(MobiComConversationFragment.class)) {
+            Utils.printLog(context, TAG, "Implement FeedbackFragmentListener in your parent fragment.");
+            throw new ClassCastException("Implement FeedbackFragmentListener in your parent fragment.");
+        }
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.feedback_rating_layout, container, false);
+
+        editTextFeedbackComment = view.findViewById(R.id.idEditTextFeedback);
+        buttonSubmitFeedback = view.findViewById(R.id.idButtonSubmit);
+
+        buttonsRating = new Button[3];
+        textViewsRatingText = new TextView[3];
+
+        buttonsRating[0] = view.findViewById(R.id.idButtonPoor);
+        buttonsRating[1] = view.findViewById(R.id.idButtonAverage);
+        buttonsRating[2] = view.findViewById(R.id.idButtonGood);
+        textViewsRatingText[0] = view.findViewById(R.id.idTextPoor);
+        textViewsRatingText[1] = view.findViewById(R.id.idTextAverage);
+        textViewsRatingText[2] = view.findViewById(R.id.idTextGood);
+        buttonCloseFragment = view.findViewById(R.id.idCloseFeedbackFragment);
+
+        editTextFeedbackComment.setVisibility(View.GONE);
+        textViewsRatingText[0].setVisibility(View.INVISIBLE);
+        textViewsRatingText[1].setVisibility(View.INVISIBLE);
+        textViewsRatingText[2].setVisibility(View.INVISIBLE);
+
+        buttonSubmitFeedback.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(ratingLevel == 0) {
+                    setRatingLevel(RATING_AVERAGE);
+                }
+                String feedbackComment = editTextFeedbackComment.getText().toString().trim();
+                feedbackFragmentListener.onFeedbackSubmitButtonPressed(getRatingLevel(), feedbackComment);
+                getFragmentManager().popBackStack();
+            }
+        });
+
+        buttonCloseFragment.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                getFragmentManager().popBackStack();
+            }
+        });
+
+        //set the un-selected size of the buttons to 80% of size
+        buttonsRating[0].setScaleX(0.8f);
+        buttonsRating[0].setScaleY(0.8f);
+        buttonsRating[1].setScaleX(0.8f);
+        buttonsRating[1].setScaleY(0.8f);
+        buttonsRating[2].setScaleX(0.8f);
+        buttonsRating[2].setScaleY(0.8f);
+
+        buttonsRating[0].setOnClickListener(this);
```

Comment:
Something looks wrong here, if we decide to use rating on scale of 1 to 10 then do we need to write 10 click listeners and then 10 if else checks to compare the values? Not aware of android development but this is doesn't seem to be the right way programtically.

Reply:
Its just one click listener, being refereced ten times. If we need 10 values, a custom view or layout would be the better option to go with. How ever, the basic code would be more or less the same as what i have used. I could create a custom toggle view but that will take comparitively a lot more time than doing this.

Reply:
then do we need to write this 10 times
        buttonsRating[0].setOnClickListener(this);
        buttonsRating[1].setOnClickListener(this);
        buttonsRating[2].setOnClickListener(this);
        buttonsRating[3].setOnClickListener(this);
.....
... and so on

and then another 10 conditions
        else if (buttonId == R.id.idButtonAverage) {  //for rating 1 
        else if (buttonId == R.id.idButtonAverage) { // for rating 2..
for rating 3...
and so on to rating 10...

doesn't look correct, please find a better way to accomplish this.